pkg/controller/resourcegraphdefinition/controller.go

```diff
@@ -109,50 +106,18 @@ func (r *ResourceGraphDefinitionReconciler) SetupWithManager(mgr ctrl.Manager) e
 				MaxConcurrentReconciles: r.maxConcurrentReconciles,
 			},
 		).
-		WatchesMetadata(
-			&extv1.CustomResourceDefinition{},
-			handler.EnqueueRequestsFromMapFunc(r.findRGDsForCRD),
-			builder.WithPredicates(predicate.Funcs{
-				UpdateFunc: func(e event.UpdateEvent) bool {
-					return true
-				},
-				CreateFunc: func(e event.CreateEvent) bool {
-					return false
-				},
-				DeleteFunc: func(e event.DeleteEvent) bool {
-					return false
-				},
-			}),
-		).
-		Complete(reconcile.AsReconciler[*v1alpha1.ResourceGraphDefinition](mgr.GetClient(), r))
-}
-
-// findRGDsForCRD returns a list of reconcile requests for the ResourceGraphDefinition
-// that owns the given CRD. It is used to trigger reconciliation when a CRD is updated.
-func (r *ResourceGraphDefinitionReconciler) findRGDsForCRD(ctx context.Context, obj client.Object) []reconcile.Request {
-	mobj, err := meta.Accessor(obj)
-	if err != nil {
-		return nil
-	}
-
-	// Check if the CRD is owned by a ResourceGraphDefinition
-	if !metadata.IsKROOwned(mobj) {
-		return nil
-	}
-
-	rgdName, ok := mobj.GetLabels()[metadata.ResourceGraphDefinitionNameLabel]
-	if !ok {
-		return nil
-	}
-
-	// Return a reconcile request for the corresponding RGD
-	return []reconcile.Request{
-		{
-			NamespacedName: types.NamespacedName{
-				Name: rgdName,
+		Owns(&extv1.CustomResourceDefinition{}, builder.WithPredicates(predicate.Funcs{
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				return true
 			},
-		},
-	}
+			CreateFunc: func(e event.CreateEvent) bool {
+				return false
+			},
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				return false
```

Comment:
I would say we do want to reconcile RGDs when the CRD they manage gets deleted so we re-create them